src/main/java/net/dv8tion/jda/core/JDABuilder.java

```diff
@@ -134,6 +165,11 @@ public JDABuilder setReconnectQueue(SessionReconnectQueue queue)
     public JDABuilder setToken(String token)
     {
         this.token = token;
+        //Share ratelimit for the same token
+        // when this builder is used to build different accounts this makes sure we don't use the
+        // same ratelimiter on them as it would be inaccurate
+        if (accountType == AccountType.BOT)
+            shardRateLimiter = new ShardedRateLimiter();
```

Comment:
Check if the previous token is the same as the new one before deciding to change. If they are the same, you shouldn't be making a new ratelimiter.